main.py

```diff
@@ -202,8 +208,8 @@ def _load_switch(self, switch_id, switch_att):
 
         intf_ids = [v["id"] for v in switch_att.get("interfaces", {}).values()]
         intf_details = self.topo_controller.get_interfaces_details(intf_ids)
-        with self._links_lock:
-            self.load_interfaces_available_tags(switch, intf_details)
+        with self._interface_lock:
```

Comment:
If you could revert here to use `self._links_lock`, even `self._links_lock` was probably not completely needed, but let's keep the same behavior in this part. Here it's also only executed in the Main thread so it's not expected to result in any issues, in the future we can try to remove `self._links_lock` specifically from this part too.

Reply:
Reverted lock. Deleted `self._links_lock` this case was its only use case.